doc/porting_processing.dox

```diff
@@ -0,0 +1,72 @@
+- Derive your algorithms from the new base class QgsProcessingAlgorithm (or a subclass of QgsProcessingAlgorithm), not GeoAlgorithm
+
+- Ensure that your algorithm (or algorithm's parent class) implements the new pure virtual createInstance(self, config) call to return a new instance of the algorithm class. Note that this should not return a copy of the algorithm, but instead a newly constructed instance of the class. If you use a base class in your plugin for all algorithms, you can usually shortcut the createInstance implementation by placing:
+ 
+    def createInstance(self, config={}):
+        return type(self)()
+
+  inside your algorithm base class.
+
+- Input parameters and available outputs must be declared in an implementation of the new pure virtual method initAlgorithm(self, config={})
+
+- The input parameters and outputs classes have been replaced with new c++ versions, which must be used when calling addParameter and addOuput.    
+Map parameters from old classes to new classes
+   - ParameterNumber -> QgsProcessingParameterNumber
+       - constructor arguments in different order
+	   - type specified explicitly, not via min/max/default type
+   - ParameterBoolean -> QgsProcessingParameterBoolean
+   - ParameterCrs -> QgsProcessingParameterCrs
+
+- Use featuresources and sinks wherever possible
```

Comment:
I would add some hints here to help decide when one should use sinks/sources and when to use layers